src/dosbox.cpp

```diff
@@ -624,20 +619,42 @@ void DOSBOX_Init(void) {
 	};
 	pstring->Set_values(midi_devices);
 	pstring->Set_help(
-	        "Device that will receive the MIDI data from emulated MPU-401.\n"
+	        "Device that will receive the MIDI data (from the emulated MIDI\n"
+	        "interface - MPU-401).\n"
 #if C_FLUIDSYNTH
 	        "Use 'fluidsynth' to select built-in software synthesiser,\n"
 	        "see the fluidsynth section for detailed configuration.\n"
 #endif
-	        "Use 'auto' to pick the first working device.");
-
-	Pstring = secprop->Add_string("midiconfig",Property::Changeable::WhenIdle,"");
-	Pstring->Set_help("Special configuration options for the device driver. This is usually the id or part of the name of the device you want to use\n"
-	                  "(find the id/name with mixer/listmidi).\n"
-	                  "Or in the case of coreaudio, you can specify a soundfont here.\n"
-	                  "When using a Roland MT-32 rev. 0 as midi output device, some games may require a delay in order to prevent 'buffer overflow' issues.\n"
-	                  "In that case, add 'delaysysex', for example: midiconfig=2 delaysysex\n"
-	                  "See the README/Manual for more details.");
+	        "Use 'auto' to pick the first working external device.");
+
+	pstring = secprop->Add_string("midiconfig", when_idle, "");
+	pstring->Set_help(
+	        "Special configuration options for the device driver.\n"
+	        "This is usually the id or part of the name of the device you\n"
+	        "want to use (find the id/name with DOS command 'mixer /listmidi').\n"
```

Comment:
I like the last sentence describing how to how discover the value by running the mixer command.

The first sentence adds more confusion, so we should try to reword it. 

- What's so "special" about this configuration value versus other non-special conf values? (we should drop "Special").

- It mentions a "Device driver", yet I believe all of the MIDI players are software (they're not hardware nor are there device-drivers for the player). 

My understanding is that this value tells DOSBox how to interface with selected MIDI player. Is that right?

Reply:
There's nothing special about it, I agree.

MIDI players can be hardware (e.g. external keyboard plugged in or modern sound card (however rare they are)).

This value behaves differently depending on what mididevice is selected - usually it selects which available MIDI device to use.

Reply:
>  depending on what mididevice is selected - usually it selects which available MIDI device to use.

There is some circular logic here. (_Depending on the midi device, it selects the midi device_).

Dosbox can send MIDI sequences through a selected **interface** (ALSA, CoreAudio, and WindowsMIDI/win32), which may play the MIDI itself in the case of CoreAudio (when a path to an SF2 is provided), or route the sequence to one or more players (physical mt32, Qsynth, Timidity, munt, ...) that are connected to the interface, via a **connection identifier**.

We want the user to choose their _MIDI interface_ and either _connection identifier_ or _SF2_ in the case of CoreAudio.

Ideally we could rename these settings.. but that would hurt backward compatibility. Maybe we can clarify the descriptions.